### PR TITLE
DistilBERT text classification pipeline

### PR DESCRIPTION
**Pull request by Yorko**
## Description

Text classification pipeline with DistilBERT from [transformers](https://github.com/huggingface/transformers) and Catalyst. 
This PR sticks to Config API. The same within Jupyter API is presented in [this Kaggle Kernel](https://www.kaggle.com/kashnitsky/distillbert-catalyst-amazon-product-reviews). A tutorial based on this Kernel will be contributed as a separate PR. 

To run the example, got to `examples` and run `catalyst-dl run -C distilbert_text_classification/config.yml`, read the Readme file in `distilbert_text_classification` as well. 

## Related Issue

No related issues, but this PR is mostly based on the work done by xelibrion and lightforever in branch [`nlp/bert_ner`](https://github.com/catalyst-team/catalyst/tree/nlp/bert_ner).

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [Code of Conduct](https://github.com/catalyst-team/catalyst/blob/master/CODE_OF_CONDUCT.md) document.
- [x] I have read the [Contributing](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md) guide.
- [x] I have checked the code-style using `make check-style`.
- [x] I have written the docstring in Google format for all the methods and classes that I used.
- [x] I have checked the docs using `make check-docs`.

### CODE REVIEW

**Comment by Yorko** [comment]
`./bin/_check_codestyle.sh -s` locally succeeds (48 passed, 10 warnings in 4.64 seconds), but fails with Travis 🤔

File: .travis.yml [hunk 1]
```diff
@@ -33,11 +33,15 @@ install_dl: &requirements_dl
 install_rl: &requirements_all
   pip install -r requirements/requirements.txt -r requirements/requirements-rl.txt
 
+install_nlp: &requirements_nlp
+  pip install -r requirements/requirements-nlp.txt
```
Comment (Yorko) [inline .travis.yml:34]: dunno, whether it needs to go to &requirements_all as well.

File: .travis.yml [hunk 2]
```diff
@@ -33,11 +33,15 @@ install_dl: &requirements_dl
 install_rl: &requirements_all
```
Comment (Yorko) [inline .travis.yml]: A bit confusing why install_rl: has an annotation &requirements_all.
Feel free to refactor the way I added NLP requirements.

File: catalyst/data/nlp/classify.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import logging
+from typing import List, Mapping
+
+from transformers import AutoTokenizer
+
+import torch
+from torch.utils.data import Dataset
+
+
+class TextClassificationDataset(Dataset):
```
Comment (Yorko) [inline catalyst/data/nlp/classify.py:9]: This class is tested in `catalyst/data/nlp/tests/test_text_clf_dataset.py`. Tests can be extended of course.

File: catalyst/dl/runner/supervised.py [hunk 4]
```diff
@@ -26,10 +27,10 @@ class SupervisedRunner(Runner):
 
     def __init__(
```
Comment (Yorko) [inline catalyst/dl/runner/supervised.py]: This part is actually inherited from the `nlp/bert_ner` branch.

File: catalyst/utils/dataset.py [hunk 5]
```diff
@@ -53,9 +53,8 @@ def create_dataset(
     return dataset
 
 
-def split_dataset(
-    dataset: pd.DataFrame, **train_test_split_args
-) -> Tuple[DictDataset, DictDataset]:
+def split_dataset(dataset: pd.DataFrame,
```
Comment (Yorko) [inline catalyst/utils/dataset.py]: This and several more files are automatically refactored by code style checker.

**Comment by Yorko** [comment]
@Scitator @lightforever @TezRomacH now it's ready for review, added some comments to code. Feel free to ping me on Slack. Can have a short discussion live (on Slack).

**Comment by Scitator** [comment]
@Yorko  could you please merge with master? we have fixed the CI tests issue

**Comment by Yorko** [comment]
@Scitator done!

**Comment by Yorko** [comment]
@Scitator if I pull changes from the current master, `make check-style` [fails](https://travis-ci.com/catalyst-team/catalyst/jobs/262772070), locally as well

**Comment by lightforever** [comment]
@Yorko, please merge with master and ensure you have the libraries in the list: https://github.com/catalyst-team/catalyst/blob/master/requirements/requirements-dev.txt

The versions must be the same. Especially, yapf must have version 0.28.0

**Pull request merged**